proxy/src/outbound.rs

```diff
@@ -92,38 +79,46 @@ where
         // by `NormalizeUri`, as we need to know whether the request will
         // be routed by Host/authority or by SO_ORIGINAL_DST, in order to
         // determine whether the service is reusable.
-        let local = req.uri().authority_part()
+        let authority = req.uri().authority_part()
             .cloned()
         // Therefore, we need to check the host header as well as the URI
         // for a valid authority, before we fall back to SO_ORIGINAL_DST.
-            .or_else(|| h1::authority_from_host(req))
-            .map(|authority| {
-                FullyQualifiedAuthority::normalize(
-                    &authority,
-                    &self.default_namespace)
-            });
-
-        // If we can't fully qualify the authority as a local service,
-        // and there is no original dst, then we have nothing! In that
-        // case, we return `None`, which results an "unrecognized" error.
-        //
-        // In practice, this shouldn't ever happen, since we expect the proxy
-        // to be run on Linux servers, with iptables setup, so there should
-        // always be an original destination.
-        let dest = if let Some(NamedAddress {
-            name,
-            use_destination_service: true
-        }) = local {
-            Destination::LocalSvc(name)
-        } else {
-            let orig_dst = req.extensions()
+            .or_else(|| h1::authority_from_host(req));
+
+
+        // TODO: Return error when try_from() fails.
```

Comment:
Which `try_from`? I don't see one inside this match statement, is it actually inside one of the functions?

Reply:
> Which try_from? I don't see one inside this match statement, is it actually inside one of the functions?

I updated the comment to refer to `HostAndPort::normalize()`. If there is a request-target value but that value is invalid, then we shouldn't route to `SO_ORIGINAL_DST`; we should instead fail because the request is invalid.